explore/explore_internal.h

```diff
@@ -340,8 +341,8 @@ namespace exploration
   // 2) Does not normalize the scores (unlike sample_after_normalization)
   // 3) Scores need not add up to one.
   template <typename It>
-  int sample_scores(
-      uint64_t* p_seed, It scores_first, It scores_last, uint32_t& chosen_index, std::random_access_iterator_tag)
+  int sample_scores(std::shared_ptr<rand_state> p_seed, It scores_first, It scores_last, uint32_t& chosen_index,
```

Comment:
This function is part of the explore library which is independent, and therefore cannot depend on `rand_state`. The inner random state should be retrieved before this is called, then it should be updated with the updated state once `sample_scores` is complete.

Reply:
Changed to get seed in sample_pdf before calling this